rskj-core/src/main/java/co/rsk/core/bc/BlockExecutor.java

```diff
@@ -265,97 +447,147 @@ private BlockResult executeInternal(
-                    tx,
-                    txindex++,
-                    block.getCoinbase(),
-                    track,
-                    block,
-                    totalGasUsed,
-                    vmTrace,
-                    deletedAccounts);
-            boolean transactionExecuted = txExecutor.executeTransaction();
+        TransactionsPartitioner partitioner2 = new TransactionsPartitioner();
+        TransactionsPartitionExecutor partExecutor = partitioner2.newPartitionExecutor(partitioner2.newPartition());
 
+        TransactionConflictDetector transactionConflictDetector = new TransactionConflictDetector(partitioner2);
+        if (track.isCached()) {
+            // track the repository to detect conflicts
+            track.getCacheTracking().subscribe(transactionConflictDetector);
+        }
 
-            if (!acceptInvalidTransactions && !transactionExecuted) {
-                if (discardInvalidTxs) {
-                    logger.warn("block: [{}] discarded tx: [{}]", block.getNumber(), tx.getHash());
-                    continue;
-                } else {
-                    logger.warn("block: [{}] execution interrupted because of invalid tx: [{}]",
-                                block.getNumber(), tx.getHash());
-                    profiler.stop(metric);
-                    return BlockResult.INTERRUPTED_EXECUTION_BLOCK_RESULT;
-                }
-            }
+        int txindex = 0;
 
-            executedTransactions.add(tx);
+        for (Transaction tx : block.getTransactionsList()) {
 
-            if (vmTrace) {
-                txExecutor.extractTrace(programTraceProcessor);
+            if (txindex == nextIndexPartition + 1) {
+                // start a new partition from this transaction
+                nextIndexPartition = itPartitionEnds.hasNext() ? itPartitionEnds.next() : -2;
+                partExecutor = partitioner2.newPartitionExecutor(partitioner2.newPartition());
             }
 
-            logger.trace("tx executed");
+            TransactionExecutorTask txExecutorTask = new TransactionExecutorTask(
+                    transactionExecutorFactory,
+                    tx,
+                    txindex,
+                    block,
+                    blockSharedData,
+                    acceptInvalidTransactions,
+                    discardInvalidTxs);
 
-            // No need to commit the changes here. track.commit();
+            partExecutor.addTransactionTask(txExecutorTask);
 
-            logger.trace("track commit");
+            txindex++;
+        }
 
-            long gasUsed = txExecutor.getGasUsed();
-            totalGasUsed += gasUsed;
-            Coin paidFees = txExecutor.getPaidFees();
-            if (paidFees != null) {
-                totalPaidFees = totalPaidFees.add(paidFees);
+        // Wait for all thread terminate
+        try {
+            partitioner2.waitForAllThreadTermination(
+                    10000,
```

Comment:
Magic number should be a pre-defined constant

Reply:
Sure !
Done.